### ISSUE TITLE
Rename FeatureAugmenter

### ISSUE THREAD

**Issue by jovan-stojanovic** [skrub-data/skrub#628]
### Problem Description

Renaming FeatureAugmenter to Joiner and AggregationJoin #600 to AggregationJoiner so as to make consistent naming of features.

### Feature Description

Renaming in all files, links, docs, examples etc.

**Comment by Vincent-Maladiere** [comment]
I feel like `AggJoiner` is explicit enough and shorter, WDYT? :)

**Comment by jovan-stojanovic** [comment]
I agree, #674 proposes renaming the `FeatureAugmenter` to `Joiner` so that's compatible I think.

**Comment by Vincent-Maladiere** [comment]
Nice, I need to take a look at it

**Comment by LilianBoulard** [comment]
> I feel like `AggJoiner` is explicit enough

Mmm... is it?
I don't know, but I dislike abbreviating things like that, I don't think we live in an age anymore where names need to be so short.

**Comment by GaelVaroquaux** [comment]
> I don't know, but I dislike abbreviating things like that, I don't think we live in an age anymore where names need to be so short.

Beginners struggle with long names: they do not know how to use their environment to complete names, and they struggle formatting long lines (they do not have the discipline to put line returns in the right place).

A lot of our target audience (data scientist) suffer from these limitations.

**Comment by LilianBoulard** [comment]
If misplacement of line breaks is a problem of code readability, I think beginners suffer far more from reading code with lots of abbreviations than having to scroll horizontally :)
But I agree Jupyter Notebooks are still missing key features like linting and good auto-completion. Still, I think readability trumps all :smile:

**Comment by Vincent-Maladiere** [comment]
In the `AggJoiner` defense, also note that the "agg" abbreviation is now widely used in data science: [Pandas uses it](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.agg.html), [Polars uses it](https://pola-rs.github.io/polars/py-polars/html/reference/dataframe/api/polars.dataframe.groupby.GroupBy.agg.html), even [Spark uses it](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.agg.html).

**Comment by GaelVaroquaux** [comment]
> In the AggJoiner defense, also note that the "agg" abbreviation is now widely used in data science: Pandas uses it, Polars uses it, even Spark uses it.

+1

**Comment by LilianBoulard** [comment]
> In the AggJoiner defense, also note that the "agg" abbreviation is now widely used in data science: [Pandas uses it](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.agg.html), [Polars uses it](https://pola-rs.github.io/polars/py-polars/html/reference/dataframe/api/polars.dataframe.groupby.GroupBy.agg.html), even [Spark uses it](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.agg.html).

Okay, good to know, let's do it then!
